orttraining/orttraining/training_api/utils.cc

```diff
@@ -63,11 +63,18 @@ Status OrtValueLike(const SessionState& sess_state, const OrtValue& input_val, O
   auto element_type = param_tensor.DataType();
   auto p_tensor = std::make_unique<Tensor>(element_type, shape, allocator);
 
-  // TODO: handle CUDA memset
   if (tensor_location.device.Type() == OrtDevice::CPU ||
       tensor_location.mem_type == OrtMemTypeCPUInput ||
       tensor_location.mem_type == OrtMemTypeCPUOutput) {
     memset(p_tensor->MutableDataRaw(), 0, p_tensor->SizeInBytes());
+  } else {
```

Comment:
We can check for whether the tensor can be copied from src (cpu) to dest. What benefit do we get by checking tensor location is on cuda? We probably don't want to put a constraint on that, right?

Reply:
On reading the code more, the check whether the data transfer can copy the data or not already happens inside `DataTransferManager::CopyTensor`. So, it might not be needed here.